src/core/caniuse.js

```diff
@@ -0,0 +1,206 @@
+/*
+Module: "core/caniuse"
+Adds a caniuse support table for a "feature" #1238
+
+`conf.caniuse = { feature: key }` =>
+  .. the table is added before Copyright
+Optional settings:
+  `conf.caniuse.browsers` list of caniuse supported browser names
+    to be shown in the table or "ALL"
+    default: ["chrome", "firefox", "safari", "edge"]
+  `conf.caniuse.versions` number of browser versions to show
+  `conf.caniuse.maxAge` (in ms) local response cache duration
+  `conf.caniuse.apiURL` URL from where to fetch stats.
+    use {FEATURE} as placeholder in URL to replace it by a feature name
+*/
+
+import { semverCompare } from "core/utils";
+import IDBCache from "core/idb-cache";
+import { pub } from "core/pubsubhub";
+import "deps/hyperhtml";
+import caniuseCss from "deps/text!core/css/caniuse.css";
+
+const BROWSERS = new Map([ // browser name dictionary
+  ["chrome", "Chrome"],
+  ["firefox", "Firefox"],
+  ["ie", "IE"],
+  ["edge", "Edge"],
+  ["android", "Android"],
+  ["safari", "Safari"],
+  ["opera", "Opera"],
+  ["bb", "Blackberry"],
+  ["and_uc", "UC (Android)"],
+  ["and_ff", "Firefox (Android)"],
+]);
+
+export const name = "core/caniuse";
+
+export async function run(conf, doc, cb) {
+  cb(); // carry on with critical plugins
+  if (!conf.caniuse) {
+    return;
+  }
+  normalizeConf(conf);
+  const { caniuse } = conf;
+  if (!caniuse.feature) {
+    return; // no feature to show
+  }
+  // Opportunistically insert the style into the head to reduce FOUC.
+  document.head.appendChild(hyperHTML `<style>${caniuseCss}</style>`);
+
```

Comment:
Another thing we want to do here is insert a preconnect:

```JS
import { createResourceHint } from "core/utils";

const link = createResourceHint({
  hint: "preconnect",
  href: "https://raw.githubusercontent.com",
}); 
document.head.appendChild(link);
```

This will warm up the connection for us, ready to download the JSON.